### PR TITLE
Add db_instance acceptance tests & sweepers

### PR DESCRIPTION
**Pull request by trevorbonas**
### Description

- Acceptance tests have been added for `aws_timestreaminfluxdb_db_instance`.
- Sweepers have been added for `aws_timestreaminfluxdb_db_instance`.

### Relations

N/A

### References

N/A

### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccTimestreamInfluxDBDbInstance PKG=timestreaminfluxdb ACCTEST_PARALLELISM=1
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.2 test ./internal/service/timestreaminfluxdb/... -v -count 1 -parallel 1 -run='TestAccTimestreamInfluxDBDbInstance'  -timeout 360m
=== RUN   TestAccTimestreamInfluxDBDbInstance_basic
=== PAUSE TestAccTimestreamInfluxDBDbInstance_basic
=== RUN   TestAccTimestreamInfluxDBDbInstance_disappears
=== PAUSE TestAccTimestreamInfluxDBDbInstance_disappears
=== RUN   TestAccTimestreamInfluxDBDbInstance_logDeliveryConfiguration
=== PAUSE TestAccTimestreamInfluxDBDbInstance_logDeliveryConfiguration
=== RUN   TestAccTimestreamInfluxDBDbInstance_publiclyAccessible
=== PAUSE TestAccTimestreamInfluxDBDbInstance_publiclyAccessible
=== RUN   TestAccTimestreamInfluxDBDbInstance_deploymentTypeMultiAzStandby
=== PAUSE TestAccTimestreamInfluxDBDbInstance_deploymentTypeMultiAzStandby
=== RUN   TestAccTimestreamInfluxDBDbInstance_username
=== PAUSE TestAccTimestreamInfluxDBDbInstance_username
=== RUN   TestAccTimestreamInfluxDBDbInstance_bucket
=== PAUSE TestAccTimestreamInfluxDBDbInstance_bucket
=== RUN   TestAccTimestreamInfluxDBDbInstance_organization
=== PAUSE TestAccTimestreamInfluxDBDbInstance_organization
=== RUN   TestAccTimestreamInfluxDBDbInstance_tags
=== PAUSE TestAccTimestreamInfluxDBDbInstance_tags
=== RUN   TestAccTimestreamInfluxDBDbInstance_dbInstanceType
=== PAUSE TestAccTimestreamInfluxDBDbInstance_dbInstanceType
=== CONT  TestAccTimestreamInfluxDBDbInstance_basic
--- PASS: TestAccTimestreamInfluxDBDbInstance_basic (759.81s)
=== CONT  TestAccTimestreamInfluxDBDbInstance_username
--- PASS: TestAccTimestreamInfluxDBDbInstance_username (708.80s)
=== CONT  TestAccTimestreamInfluxDBDbInstance_dbInstanceType
--- PASS: TestAccTimestreamInfluxDBDbInstance_dbInstanceType (5269.20s)
=== CONT  TestAccTimestreamInfluxDBDbInstance_tags
--- PASS: TestAccTimestreamInfluxDBDbInstance_tags (724.22s)
=== CONT  TestAccTimestreamInfluxDBDbInstance_organization
--- PASS: TestAccTimestreamInfluxDBDbInstance_organization (864.17s)
=== CONT  TestAccTimestreamInfluxDBDbInstance_bucket
--- PASS: TestAccTimestreamInfluxDBDbInstance_bucket (663.05s)
=== CONT  TestAccTimestreamInfluxDBDbInstance_publiclyAccessible
--- PASS: TestAccTimestreamInfluxDBDbInstance_publiclyAccessible (862.22s)
=== CONT  TestAccTimestreamInfluxDBDbInstance_deploymentTypeMultiAzStandby
--- PASS: TestAccTimestreamInfluxDBDbInstance_deploymentTypeMultiAzStandby (1109.05s)
=== CONT  TestAccTimestreamInfluxDBDbInstance_logDeliveryConfiguration
--- PASS: TestAccTimestreamInfluxDBDbInstance_logDeliveryConfiguration (982.18s)
=== CONT  TestAccTimestreamInfluxDBDbInstance_disappears
--- PASS: TestAccTimestreamInfluxDBDbInstance_disappears (732.27s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/timestreaminfluxdb 12679.038s
```

### CODE REVIEW

**Comment by trevorbonas** [comment]
Note that the acceptance tests need `ACCTEST_PARALLELISM=1`, since Timestream for InfluxDB doesn't like more than one instance being created at a time.

File: internal/service/timestreaminfluxdb/db_instance_test.go [hunk 1]
```diff
@@ -295,63 +545,428 @@ func testAccPreCheck(ctx context.Context, t *testing.T) {
...
+		}
+	}
+}
+`, rName))
+}
+
+// Configuration with log_delivery_configuration set but not enabled.
+func testAccDbInstanceConfig_logDeliveryConfigurationNotEnabled(rName string) string {
+	return acctest.ConfigCompose(testAccDbInstanceConfig_base(), fmt.Sprintf(`
+resource "aws_s3_bucket" "test_s3_bucket" {
```
Comment (forestmvey) [inline internal/service/timestreaminfluxdb/db_instance_test.go:637]: Will these tests create the bucket `aws_s3_bucket`? Will this work if multiple people run the tests and there is a conflict with bucket names?

Comment (trevorbonas) [reply internal/service/timestreaminfluxdb/db_instance_test.go:637]: The bucket will have a random name (the value in `rName`) in AWS. The bucket will be referenced locally, in Terraform, as `aws_s3_bucket.test_s3_bucket`.

**Review by forestmvey — APPROVED** [review]
Looks good

**Pull request merged**